overlord/devicestate/handlers_install.go

```diff
@@ -1028,6 +1040,12 @@ func (m *DeviceManager) doInstallFinish(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	if trustedInstallObserver != nil {
+		if err := trustedInstallObserver.ObserveExistingTrustedRecoveryAssets(boot.InitramfsUbuntuSeedDir); err != nil {
+			return fmt.Errorf("cannot observe existing trusted recovery assets: %v", err)
+		}
+	}
```

Comment:
Are efi boot variables changed by this call? Would this change need to land in pc-kernel before changes to pc-gadget for installs from initramfs to work?

Reply:
shim has fallback. So if you do install in initrd with and older kernel, then it is fine, it will just need to do a quick reboot on the next boot to set the variables.

However, you will need to have b248f6c7f70a2fed0cff94e6d916cf7314d9694d in your kernel snap initrd to be able to do an installation in initrd at all. That means snap-bootstrap from snapd 2.62 minimum.

Reply:
And yes, we should not update pc-gadget on 22 and 20 until we have pc-kernel for sure updated there. But 24 should be fine.